libs/state/src/lib/rx-state.spec.ts

```diff
@@ -95,11 +95,86 @@ describe(rxState, () => {
     fixture.destroy();
     expect(RxState.prototype.ngOnDestroy).toHaveBeenCalled();
   });
+
+  describe('signals', () => {
+    it('should be undefined when key is undefined', () => {
+      const { component } = setupComponent<{ count: number }>();
+      const state = component.state;
+      const count = state.signal('count');
+
+      expect(count()).toBe(undefined);
+    });
+
+    it('should create one signal per key', () => {
+      const { component } = setupComponent<{ count: number }>(({ set }) => {
+        set({ count: 1337 });
+      });
+      const state = component.state;
+      const count = state.signal('count');
+      const count2 = state.signal('count');
+
+      expect(isSignal(count)).toBe(true);
+      expect(count === count2).toBe(true);
+    });
+
+    it('signal should get updated', () => {
+      const { component } = setupComponent<{ count: number }>(({ set }) => {
+        set({ count: 1337 });
+      });
+      const state = component.state;
+      const count = state.signal('count');
+      expect(count()).toBe(1337);
+
+      state.set({ count: 1 });
+      expect(count()).toBe(1);
+
+      state.connect(of({ count: 2 }));
+      expect(count()).toBe(2);
+    });
+
+    xit('should connect a signal', () => {
+      // TODO: we need TestBed flushEffect for it
```

Comment:
Instead of using flushEffect we can call the signal in cmp template and run detectChanges, it will act the same as the templates act as effects.